pkg/credentialprovider/config.go

```diff
@@ -73,11 +73,23 @@ func GetPreferredDockercfgPath() string {
 	defer preferredPathLock.Unlock()
 	return preferredPath
 }
+func GetDockercfgJsonPaths() []string {
+	return []string{GetPreferredDockercfgPath(), workingDirPath, homeJsonDirPath, rootJsonDirPath}
+}
+func GetDockercfgFilePaths() []string {
+	return []string{GetPreferredDockercfgPath(), workingDirPath, homeDirPath, rootDirPath}
+}
+
+func ReadDockerConfigFile(paths []string) (cfg DockerConfig, err error) {
```

Comment:
hmm... a single set of paths doesn't intersect well when the function is parsing two different ways (readDockerConfigJsonFileFromBytes and readDockerConfigFileFromBytes). do we need a set of jsonconfigfile and configfile paths? that's unwieldy